mutation/mutation_partition.hh

```diff
@@ -1165,6 +1165,29 @@ struct apply_resume {
     }
 };
 
+[[noreturn]] void on_bad_row_key(const schema& s, position_in_partition_view pos, const char* reason);
+
+inline void check_row_key(const schema& s, const clustering_key& key, is_dummy dummy) {
+    if (!dummy && !s.is_compact_table() && !key.is_full(s)) {
```

Comment:
It's better to start with !key.is_full(), since it will be very common for this test to fail, and the rest of the code will be cold.

Reply:
(on the other hand, it's common for !s.is_compact_table() and !dummy to be true, so we'll just burn instructions to get to !key.is_full())

Reply:
It's still better to put the failing tests before the passing tests, even if they are more expensive.

Do we need is_full? AFAIK all that was observed was a completely empty key. If it was caused by a moved-from object, that's enough.

Reply:
Prefix keys are just as bad. We haven't seen any in the wild, but while at it we might as well cover all the cases.